src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -354,6 +358,39 @@ public boolean indexExists(String index) {
     );
   }
 
+  /**
+   * Checks whether the alias exists.
+   *
+   * @param alias the alias to check
+   * @return true if it exists, false if it does not
+   */
+  public boolean aliasExists(String alias) {
+    GetAliasesRequest request = new GetAliasesRequest(alias);
+    return callWithRetries(
+        "check if alias " + alias + " exists",
+        () -> client.indices().existsAlias(request, RequestOptions.DEFAULT)
+    );
+  }
+
+  /**
+   * Resolves input target to an actual index. Indexes and data streams are returned as-is,
+   * while aliases are resolved to write index.
+   *
+   * @param target the target to check
+   * @return actual index name
+   * @throws ConnectException if alias does not have write index
+   */
+  public String resolveIndexIfAlias(String target) {
+    if (config.isDataStream()) {
+      return target;
+    }
+    String actualIndex = aliasWriteIndex(target);
```

Comment:
If the target is not an alias, what will this return?

Reply:
Yes, ElasticSearch API will return the same index name if the target is not an alias since the target is sent as an index query parameter in the GetAliasesRequest. 
Basically:
- if the target is an index, _alias API will return only one entry
- if the target is an alias, _alias API will return related index entries